### PR TITLE
Removed wrong inheritance of SCDP and SCBR from Disk class

### PR DESCRIPTION
**Pull request by uweseimet**
Besides repairing the broken inheritance (this fixes issues 189 and 663) I also fixed some TODOs.

### CODE REVIEW

**Comment by uweseimet** [comment]
I have to revert this PR to draft status because I may have found a bug.

**Comment by uweseimet** [comment]
Resolving the issue I found resulted in also resolving issue #189.

File: cpp/hal/gpiobus.cpp [hunk 1]
```diff
@@ -981,7 +981,7 @@ int GPIOBUS::GetCommandByteCount(BYTE opcode)
         return 16;
     } else if (opcode == 0xA0) {
         return 12;
-    } else if (opcode == 0x05 || (opcode >= 0x20 && opcode <= 0x7D)) {
```
Comment (akuker) [inline cpp/hal/gpiobus.cpp:984]: @uweseimet - What is opcode 5?

Comment (uweseimet) [reply cpp/hal/gpiobus.cpp:984]: @akuker I don't know, this is why I removed it. It might be something used in the past by the SASI controller.

**Comment by akuker** [comment]
@uweseimet - one question, then this is ready to approve

**Review by akuker — APPROVED** [review]

**Pull request merged**